app/dao/services_dao.py

```diff
@@ -325,27 +325,48 @@ def dao_fetch_todays_stats_for_all_services(include_from_test_key=True):
 @statsd(namespace='dao')
 def fetch_stats_by_date_range_for_all_services(start_date, end_date, include_from_test_key=True):
     start_date = get_london_midnight_in_utc(start_date)
-    end_date = get_london_midnight_in_utc(end_date)
-    end_date += timedelta(hours=23, minutes=59, seconds=59)
+    end_date = get_london_midnight_in_utc(end_date + timedelta(days=1))
+
+    if start_date >= datetime.utcnow() - timedelta(days=7):
```

Comment:
Although not a big fan of it, I think this is a place that duck typing would be useful to eliminate duplicated code, something along the lines of

    if start_date >= datetime.utcnow() - timedelta(days=7):
        NotificationTable = Notification
    else:
        NotificationTable = NotificationHistory